src/veh_interact.cpp

```diff
@@ -371,6 +378,19 @@ void veh_interact::cache_tool_availability()
                 crafting_inv.has_components( "wheel_bicycle", 1 ) ||
                 crafting_inv.has_components( "wheel_motorbike", 1 ) ||
                 crafting_inv.has_components( "wheel_small", 1 );
+
+    max_lift = std::max( { g->u.max_quality( "LIFT" ),
+                           map_selector( g->u.pos(), PICKUP_RANGE ).max_quality( "LIFT" ),
+                           vehicle_selector(g->u.pos(), 1, *veh ).max_quality( "LIFT" ) } );
+
+    // cap JACK requirements at 6000kg to support arbritrarily large vehicles
+    double qual = double( std::min( veh->total_mass(), 6000 ) * 1000 ) / TOOL_LIFT_FACTOR;
+
+    // when lifting vehicles to change wheels lever action reduces the workload
+    has_jack = max_lift >= ceil( qual / 3 ) ||
```

Comment:
Could be replaced by just giving cranes `JACK` of `LIFT * 3`, which would be wayyy less ugly than that.